service/rest.go

```diff
@@ -0,0 +1,99 @@
+package service
+
+import (
+	"net/http"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/auth"
+	"github.com/evergreen-ci/evergreen/model"
+	"github.com/evergreen-ci/gimlet"
+	"github.com/pkg/errors"
+	"github.com/urfave/negroni"
+)
+
+// restContextKey is the type used to store
+type restContextKey int
+
+const RestContext restContextKey = 0
+
+type restAPIService interface {
```

Comment:
I feel like this interface shouldn't exist, but I also can't think of a cleaner way to do this :/

Reply:
I couldn't agree more. Some day we'll fix it.